File: ktv_robot/Android/TRTC/AMEVideoRenderExample.java
package com.tencent.ame;

import android.util.Log;

import com.tencent.trtc.TRTCCloudDef;
import com.tencent.trtc.TRTCCloudListener;

public class AMEVideoRenderExample implements TRTCCloudListener.TRTCVideoRenderListener {
    @Override
    public void onRenderVideoFrame(String userId, int streamType, TRTCCloudDef.TRTCVideoFrame frame) {
        Log.d("AMEVideoRenderExample", "frame.width:" + frame.width + ", frame.height:" + frame.height);

        VideoColorFrameParseResult result = AMEVideoFrameParser.parseFromYUV(frame.data, frame.width, frame.height);
        if (result.isSuccess()) {
            Log.d("AMEVideoRenderExample", "Content: " + result.getContent());
        } else {
            Log.d("AMEVideoRenderExample", "Error: " + result.getErrMsg());
        }
    }
}

File: ktv_robot/iOS/TRTC/AMEVideoRenderExample.m
//
//  AMEVideoRenderExample.m
//
//  Created by jianguoxu on 2022/1/18.
//

#import "AMEVideoRenderExample.h"
#import "AMEVideoFrameParser.h"

@interface AMEVideoRenderExample () <TRTCVideoRenderDelegate>

@end

@implementation AMEVideoRenderExample

#pragma mark - TRTCVideoRenderDelegate

- (void)onRenderVideoFrame:(TRTCVideoFrame *)frame userId:(NSString *)userId
                streamType:(TRTCVideoStreamType)streamType {
    NSLog(@"AMEVideoRenderExample frame width: %d, height: %d", frame.width, frame.height);
    VideoColorFrameParseResult *result = [AMEVideoFrameParser parseFromYUV:frame.data width:frame.width height:frame.height];
    if (result.isSuccess) {
        NSLog(@"AMEVideoRenderExample Content: %@", result.content);
    } else {
        NSLog(@"AMEVideoRenderExample Error: %@", result.errMsg);
    }
}

@end
